api/app/signals/apps/email_integrations/rules/abstract.py

```diff
@@ -64,3 +66,19 @@ def _validate_status(self, status):
         overwrite this function in the defined Rule to add the additional status validation checks
         """
         return False
+
+    def _validate_allows_contact(self, signal):
+        """
+        Validate if the user want to be contacted and if allows_contact on feedback is False t
+        o never send ANY emails to the user
+
+        If the feature flag is False we return True to still send emails to the user.
+        """
+        if not settings.FEATURE_FLAGS.get('REPORTER_MAIL_DISABLE_CONTACT_FEEDBACK_ALLOWS_CONTACT', True):
+            return True
+
+        try:
+            return signal.feedback.filter(submitted_at__isnull=False).order_by('submitted_at').last().allows_contact
```

Comment:
Tested locally and I am wondering if we need to check on the `.last()` of the `.first()` here? Maybe this is something we should ask the business.

Reply:
If the first feedback form from the user is allowed_contact false, it will never receive a new email regarding this Signal anyway.

But if a users fills in a feedback form with allows contact true, and afterwards fills in another feedback from regarding the same issue and then sends allows contact false. It should not still send emails because the first feedback allowed for True. Therefor I had chosed to go with .last()

Reply:
That's right, maybe I am overthinking this 🤔 😄